compiler/src/iree/compiler/Utils/FlatbufferUtils.h

```diff
@@ -120,6 +120,22 @@ class FlatbufferBuilder {
   // as a shaped `vector<SIZExi8>` dense attr. The builder is left unmodified.
   DenseIntElementsAttr getBufferAttr(MLIRContext *context);
 
+  // Captures the current contents of the flatbuffer builder and returns them
+  // as a shaped `vector<SIZExi8>` dense attr. The builder is left unmodified.
+  // A 64-byte prefix structure is prepended to the buffer containing the file
+  // magic number for identification, a coarse version, and an 8 byte size
+  // (excluding the header). See iree_flatbuffer_file_header_t.
+  DenseIntElementsAttr getHeaderPrefixedBufferAttr(MLIRContext *context,
+                                                   uint32_t magic,
+                                                   uint32_t version);
+  DenseIntElementsAttr getHeaderPrefixedBufferAttr(MLIRContext *context,
+                                                   const char magic[4],
```

Comment:
pass this as `std::array` to enforce the size?

Reply:
unfortunately it comes from the flatbuffers define which is just a string literal, but I'll make this take a string and strlen it for safety.